### PR TITLE
feat(cache): Stopped generating constant directory level

### PR DESCRIPTION
**Pull request by githubjeka**
As described in #18263 FileCache generate directories with permanent names when using key prefix cache

image

This PR removed `\yii\caching\FileCache::$keyPrefix` for result of `\yii\caching\FileCache::getCacheFile()` to avoid generating constant cache directory levels

**BREAKING CHANGE:** After the Yii upgrade, the storage structure for the file cache will change. It is worth warming up the cache again if there is a logical dependency when working with the file cache.

image

| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Tests pass?   | ✔️
| Fixed issues  | #18263

### CODE REVIEW

**Review by bizley — APPROVED** [review]
LGTM

**Comment by My6UoT9** [comment]
Would be nice if a upgrade message could be add, since OP mentionend

> BREAKING CHANGE: After the Yii upgrade, the storage structure for the file cache will change. It is worth warming up the cache again if there is a logical dependency when working with the file cache.

**Comment by bizley** [comment]
This ^. Please add.

**Review by samdark — APPROVED** [review]
Please add a line to CHANGELOG. Thanks.

File: framework/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ Yii Framework 2 Change Log
 2.0.39 under development
 ------------------------
 
+- Bug #18263: Fix writing `\yii\caching\FileCache` files to the same directory when `keyPrefix` is set (githubjeka)
```
Comment (githubjeka) [inline framework/CHANGELOG.md:7]: CHANGELOG.md  updated

**Comment by samdark** [comment]
Thank you!

**Pull request merged**